lua/ibl/highlights.lua

```diff
@@ -20,7 +20,7 @@ local get = function(name)
         hl = vim.api.nvim_get_hl(0, { name = name })
     else
         ---@diagnostic disable-next-line
-        hl = vim.api.nvim_get_hl_by_name(name, true)
+        hl = vim.fn.hlexists(name) == 1 and vim.api.nvim_get_hl_by_name(name, true) or nil
```

Comment:
Resolved in latest commit. The current error seems like a bug in the lua language server. Clearly it knows that the variable isn't `nil` and doesn't complain about setting `nocombine` 🤷🏼 

![2023-10-03_21:23:31_screenshot](https://github.com/lukas-reineke/indent-blankline.nvim/assets/1591837/7cbbf8f2-b3ef-4b1d-9f54-f4df7910dddd)


To make the linter happy you could do this but it's pretty pointless:
![2023-10-03_21:24:14_screenshot](https://github.com/lukas-reineke/indent-blankline.nvim/assets/1591837/dd3e6af4-d428-4ded-a215-ec6adaa44cbf)


Also funky it complatins on `nightly` and not when testing on latest stable. Could be because of the recent improvements to the lua_ls annotations in neovim core

Reply:
this is when linting becomes tricky especially when using a language server that's relatively notorious for it's instability with some stuff 😂

Reply:
I think the problem is that `vim.empty_dict()` is not typed properly
just cast it like

```lua
hl = vim.empty_dict() --[[ @as table ]]
```

and you can also add types to the `get` function to make it clear

```lua
---@param name string
---@return table<string, any>
```